source/game.database.bmx

```diff
@@ -235,10 +232,125 @@ Type TDatabaseLoader
 			Notify "Important data is missing:  series:"+totalSeriesCount+"  movies:"+totalMoviesCount+"  news:"+totalNewsCount+"  adcontracts:"+totalContractsCount
 		EndIf
 
+		LoadDatabaseLocalizations(dbDirectory)
 
 		'fix potentially corrupt data
 		FixLoadedData()
 	End Method
+
+
+	Function LoadDatabaseLocalizations(dbDirectory:String)
+		Local langDir:String = dbDirectory+"/lang/"
+		Local dbl:TDatabaseLocalizer = GetDatabaseLocalizer()
+
+		For Local l:TLocalizationLanguage = EachIn TLocalization.languages
+			Local languageId:Int = TLocalization.GetLanguageId(l.languageCode)
+			If languageId < 0 Then Continue
+			Local file:String = langDir + l.languageCode+".xml"
+			If FileType(file) = 1
+				Local xml:TXmlHelper = TXmlHelper.Create(file)
+				Local allGlobalVars:TxmlNode = xml.FindRootChildLC("globalvariables")
+				If allGlobalVars
+					Local gvl:TLocalizationLanguage = dbl.getGlobalVariables(languageId)
+					If Not gvl
+						gvl = new TLocalizationLanguage
+						gvl.languageCode = l.languageCode
+						dbl.globalVariables.insert(languageId, gvl)
+					EndIf
+					Local varName:String
+					Local value:String
+					For Local varNode:TxmlNode = EachIn xml.GetNodeChildElements(allGlobalVars)
+						varName = varNode.getName().toLower()
+						value = varNode.getContent()
+						If varName And value Then gvl.map.insert(varName, value)
+					Next
+				EndIf
+
+				Local NO_VALUE:String = "<NO_VALUE_PRESENT>"
+				Local v:String
+
+				Local nodeAllPersons:TxmlNode
+				nodeAllPersons = xml.FindRootChildLC("persons")
+				Local personsToStore:PersonLocalization[] = new PersonLocalization[0]
```

Comment:
Wenn wir 1000 Personen dort drin hätten, würden wir 1000x den Array neu machen ..
An sich nicht viel, aber dennoch würde hier wohl eine "brl.objectlist" Sinn ergeben... das ist sozusagen ein "Managed Array".
Oder klassisch mit 100 Einträgen starten und bei Erreichen der aktuellen Größe immer "in Blöcken von X" vergrößern (man muss ja am Ende nicht mal "kürzen", da die eachin-Iteration ja über Null-Einträge hinweggeht (oder man macht dann einfach ein "exit" sobald nix mehr kommt bei einer for 0 to arr.length Schleife)

Reply:
Ich bin hier gegen übertriebene Optimierung. Dieser Code wird nur einmalig am Spielbeginn aufgerufen, dann nie wieder.

Reply:
Ich lege einmalig ein Array an und übernehme dann nur die nötigen Teile davon.